File: quiz-backend/src/old-cpp/Question.cpp
#include "Question.h"
#include <iostream>

// constructor
Question::Question() {}

// Creates a question after being provided with the question, options, and answer
Question::Question(std::string q, std::vector<std::string> options, std::string answer) {
    question_ = q;
    options_ = options;
    answer_ = answer;
}

// Returns whether the user's submission for the current question is correct
bool Question::isCorrect(std::string submission) {
    return submission == answer_;
}

std::vector<std::string> Question::getOptions() {
    return options_;
}

std::string Question::getQuestion() {
    return question_;
}

std::string Question::getCorrectAnswer() {
    return correctAnswer_;
}

File: quiz-backend/src/models/Question.py
class Question:
    def __init__(self, question, options, correct_answer):
        self.question = question
        self.correct_answer = correct_answer
        self.options = options

    def is_correct(self, submission):
        return submission == self.correct_answer

    def get_options(self):
        return self.options

    def get_question(self):
        return self.question

    def get_correct_answer(self):
        return self.correct_answer

    def to_dict(self):
        return {
            'question': self.question,
            'options': self.options,
            'correct_answer': self.correct_answer,
        }

    @classmethod
    def from_dict(cls, data):
        return cls(
            question=data.get('question'),
            options=data.get('options', []),
            correct_answer=data.get('correct_answer'),
        )
    

    